internal/graph/cached_resolver.go

```diff
@@ -140,7 +150,6 @@ func (c *CachedCheckResolver) ResolveCheck(
 	req *ResolveCheckRequest,
 ) (*ResolveCheckResponse, error) {
 	span := trace.SpanFromContext(ctx)
-	checkCacheTotalCounter.Inc()
 
 	cacheKey, err := CheckRequestCacheKey(req)
```

Comment:
**Suggestion:** Move cache key generation below to immediately above where its used. Should avoid creating the key if we're not going to use it.

Reply:
I'm not sure if we have the opportunity to move this as it's always used in this function, we will either read from the cache using it or write to the cache using it